server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/completions/Snippets.java

```diff
@@ -14,39 +14,95 @@
 
-import java.io.InputStream;
-import java.io.SequenceInputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
 
 /** This class is a provider for Cobol Snippets */
+@Slf4j
 @Singleton
-public class Snippets extends CompletionStorage {
-  private static final String SNIPPETS_FILE_PATH = "LanguageSnippets.txt";
-  private static final String SNIPPETS_IDMS_FILE_PATH = "LanguageSnippetsIDMS.txt";
-  private static final String SNIPPETS_DACO_FILE_PATH = "LanguageSnippetsDaCo.txt";
+public class Snippets extends CompletionStorage<SnippetsModel> {
+  private static final String SNIPPETS_FILE_PATH = "LanguageSnippetsCobol.json";
+  private static final String SNIPPETS_IDMS_FILE_PATH = "LanguageSnippetsIDMS.json";
+  private static final String SNIPPETS_DACO_FILE_PATH = "LanguageSnippetsDaCo.json";
+
+  @Inject
+  Snippets(SettingsService settingsService) {
+    super(settingsService);
+  }
 
   @Override
-  protected InputStream getInputStream(List<String> dialectList) {
-    if (!dialectList.isEmpty()
-        && dialectList.contains(IdmsDialect.NAME)
-        && dialectList.contains(DaCoDialect.NAME)) {
-      return new SequenceInputStream(
-          Snippets.class.getResourceAsStream(SNIPPETS_FILE_PATH),
-          new SequenceInputStream(
-              Snippets.class.getResourceAsStream(SNIPPETS_IDMS_FILE_PATH),
-              Snippets.class.getResourceAsStream(SNIPPETS_DACO_FILE_PATH)));
-    } else if (!dialectList.isEmpty() && dialectList.contains(IdmsDialect.NAME)) {
-      return new SequenceInputStream(
-          Snippets.class.getResourceAsStream(SNIPPETS_FILE_PATH),
-          Snippets.class.getResourceAsStream(SNIPPETS_IDMS_FILE_PATH));
-    } else if (!dialectList.isEmpty() && dialectList.contains(DaCoDialect.NAME)) {
-      return new SequenceInputStream(
-          Snippets.class.getResourceAsStream(SNIPPETS_FILE_PATH),
-          Snippets.class.getResourceAsStream(SNIPPETS_DACO_FILE_PATH));
-    } else return Snippets.class.getResourceAsStream(SNIPPETS_FILE_PATH);
+  protected Map<String, SnippetsModel> getInputStream(List<String> dialectList) {
+    Gson gson = new Gson();
+    Reader reader;
+    Map<String, SnippetsModel> cobolSnippets = new HashMap<>();
+    Map<String, SnippetsModel> idmsSnippets;
+    try {
+      reader = getReader(SNIPPETS_FILE_PATH);
+      cobolSnippets = getCobolSnippets(gson, reader);
+      reader.close();
+
+      if (!dialectList.isEmpty()) {
+        idmsSnippets = getMergedSnippetsModelMap(SNIPPETS_IDMS_FILE_PATH, cobolSnippets);
+        if (dialectList.size() == 2
+            || (dialectList.size() == 1 && dialectList.contains(DaCoDialect.NAME))) {
+          return getMergedSnippetsModelMap(SNIPPETS_DACO_FILE_PATH, idmsSnippets);
+        } else return idmsSnippets;
```

Comment:
Better to have one variable and one `return` statement to understand the logic and to test this method